console/v1alpha1/types_console_plugin.go

```diff
@@ -15,32 +15,120 @@ type ConsolePlugin struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
 
+	// +kubebuilder:validation:Required
+	// +required
 	Spec ConsolePluginSpec `json:"spec"`
 }
 
 // ConsolePluginSpec is the desired plugin configuration.
 type ConsolePluginSpec struct {
 	// displayName is the display name of the plugin.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +optional
 	DisplayName string `json:"displayName"`
 	// service is a Kubernetes Service that exposes the plugin using a
 	// deployment with an HTTP server. The Service must use HTTPS and
-	// service serving certificate. The console backend will proxy the
+	// Service serving certificate. The console backend will proxy the
 	// plugins assets from the Service using the service CA bundle.
+	// +kubebuilder:validation:Required
+	// +required
 	Service ConsolePluginService `json:"service"`
+	// proxy is a list of Services that the plugin needs to connect to.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Proxy ConsolePluginProxy `json:"proxy"`
 }
 
-// ConsolePluginService holds information on service that is serving
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// services is a list of in-cluster Services that the plugin
+	// will connect to. The Service must use HTTPS. Console backend
+	// exposes the following endpoint in order to proxy communication
+	// between the plugin and the Service:
+	//
+	// /api/proxy/namespace/<service-namespace>/service/<service-name>:<port-number>/<request-path>?<optional-query-parameters>
+	//
+	// Request example path:
+	//
+	// /api/proxy/namespace/helm/service/helm-charts:8443/releases?limit=10
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Services []ConsolePluginProxyService `json:"services"`
+}
+
+// ConsolePluginProxyService holds information on Service to which
+// console's backend will proxy the plugin's requests.
+type ConsolePluginProxyService struct {
+	// name of Service that the plugin needs to connect to.
```

Comment:
nit: we're using inconsistent punctuation here and below where we don't have a `.`